cherab/tools/spectroscopy/polychromator.py

```diff
@@ -0,0 +1,163 @@
+
+# Copyright 2016-2021 Euratom
+# Copyright 2016-2021 United Kingdom Atomic Energy Authority
+# Copyright 2016-2021 Centro de Investigaciones Energéticas, Medioambientales y Tecnológicas
+#
+# Licensed under the EUPL, Version 1.1 or – as soon they will be approved by the
+# European Commission - subsequent versions of the EUPL (the "Licence");
+# You may not use this work except in compliance with the Licence.
+# You may obtain a copy of the Licence at:
+#
+# https://joinup.ec.europa.eu/software/page/eupl5
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the Licence is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied.
+#
+# See the Licence for the specific language governing permissions and limitations
+# under the Licence.
+
+import numpy as np
+from raysect.optical import InterpolatedSF
+from raysect.optical.observer import RadiancePipeline0D
+
+from .instrument import SpectroscopicInstrument
+
+
+class PolychromatorFilter(InterpolatedSF):
```

Comment:
Agreed. I introduced a parent `PolychromatorFilter` class that adds `min_wavelength`, `max_wavelength`, `window` and `central_wavelength` attributes to `SpectralFunction`.  `PolychromatorFilter` also ensures that its spectral function is zero outside the filter window. `TrapezoidalFilter` inherits from `PolychromatorFilter` as you suggested.

It would be nice to be able to change the spectral bins number after initialization. If we make this a property of `PolychromatorFilter`, then we must notify the `Polychromator` when it changes. In this case we have to track the polychromators to which the filter is added. I think that this would be unnecessary complication of the code, because we can already control the number of spectral bins through the `min_bins_per_window` property of the `Polychromator` object.